src/cli/util.js

```diff
@@ -398,6 +398,12 @@ function createIgnorerFromContextOrDie(context) {
 }
 
 function eachFilename(context, patterns, callback) {
+  // workaround for fast-glob on Windows ref:
+  // https://github.com/mrmlnc/fast-glob#how-to-write-patterns-on-windows
+  if (path.sep === "\\") {
+    patterns = patterns.map(path => path.replace(/\\/g, "/"));
+  }
```

Comment:
It is bad idea, on linux I can use `\` as part of directory or file name

Reply:
That's exactly why I asked to wrap this replace into `if (path.sep === "\\")`.

Reply:
What about WSL? It is allow to use linux filenames and run commands on windows and verse vice. Yes we can try to keep backward compatibility but I think we should throw a warning on using `\` as part of glob

Reply:
No problems with WSL as far as I can see. It behaves just like 'normal' Linux. You can create a file named `\` there, but if you look at that file from Windows it won't be named `\`. The names are somehow mapped.

Please describe the scenario that you worry about. I don't understand. What exactly can go wrong? Are you simply uncomfortable about the fact that commands with `\` aren't cross-platform compatible?

Reply:
Please describe the scenario that you worry about. I don't understand. What exactly can go wrong?

Glob **always**should use `/`, please read https://github.com/micromatch/micromatch#backslashes

> To solve this, you might be inspired to do something like 'foo\\*'.replace(/\\/g, '/'), but this causes another, potentially much more serious, problem.

I can spend time investigation in detail why is dangerous, but I also prefer to trust the documentation and not make some mistakes that we are warned about.

Reply:
I think we must ask developers to use `/` for glob patterns, it is part of breaking changes

Reply:
I know exactly what they mean. See [this comment](https://github.com/prettier/prettier/pull/7587#discussion_r379938060). But I think it's okay. Not accepting `\` as a path separator on Windows is not okay. That's a trade-off.

Reply:
Hm, maybe you are right, eslint has same logic https://github.com/eslint/eslint/blob/cf38d0d939b62f3670cdd59f0143fd896fccd771/lib/cli-engine/cli-engine.js#L711

Also look at that https://github.com/eslint/eslint/blob/cf38d0d939b62f3670cdd59f0143fd896fccd771/lib/cli-engine/cli-engine.js#L708

Reply:
Maybe we should use `path.normalize` to allow using `../../foo`

Reply:
It's a horrible mess, but everyone seems to be okay with it. So I guess we should be okay too.

Reply:
Let's move tests for `normalize` to https://github.com/prettier/prettier/pull/7587

Reply:
Also we will add more tests when implement `expand directories`, so :+1: on that